src/main/java/com/ollymonger/dynmap/portals/DynmapPortalsPlugin.java

```diff
@@ -154,12 +157,25 @@ public void onBlockBreak(BlockBreakEvent event) {
                 .findFirst();
 
         portal.ifPresent(p -> {
-            event.getPlayer().sendMessage("You destroyed portal " + p.getPortalId());
+            String portalId = p.getPortalId();
+
+            event.getPlayer().sendMessage("You destroyed portal " + portalId);
 
-            this.registeredPortals.remove(p);
+            Marker marker = this.portalSet.findMarker(p.getPortalId());
```

Comment:
Delete the exclusion also - and use `portalId` here rather than getting the value again